src/app/components/rpc-transaction-request/post-conditions/fungible-post-condition-item.tsx

```diff
@@ -0,0 +1,93 @@
+import {
+  type FungiblePostConditionWire,
+  PayloadType,
+  type PayloadWire,
+  addressToString,
+} from '@stacks/transactions';
+import { Box, Stack, styled } from 'leather-styles/jsx';
+
+import { formatContractId } from '@leather.io/stacks';
+import { ItemLayout, StxAvatarIcon } from '@leather.io/ui';
+import { truncateMiddle } from '@leather.io/utils';
+
+import { ftDecimals, getSafeImageCanonicalUri } from '@app/common/stacks-utils';
+import {
+  getAmountFromPostCondition,
+  getIconStringFromPostCondition,
+  getNameFromPostCondition,
+  getPostConditionCodeMessage,
+  getPostConditionTitle,
+  getSymbolFromPostCondition,
+} from '@app/common/transactions/stacks/post-condition.utils';
+import { StacksAssetAvatar } from '@app/components/stacks-asset-avatar';
+import { useCurrentStacksAccount } from '@app/store/accounts/blockchain/stacks/stacks-account.hooks';
+import { useAssetFromFungiblePostCondition } from '@app/store/transactions/post-conditions.hooks';
+
+interface FungiblePostConditionItemProps {
+  isLast?: boolean;
+  payload: PayloadWire;
+  pc: FungiblePostConditionWire;
+}
+export function FungiblePostConditionItem({ isLast, payload, pc }: FungiblePostConditionItemProps) {
+  const currentAccount = useCurrentStacksAccount();
+  const asset = useAssetFromFungiblePostCondition(pc);
+
+  const imageCanonicalUri =
+    asset?.image_canonical_uri &&
+    asset.name &&
+    getSafeImageCanonicalUri(asset.image_canonical_uri, asset.name);
+  const title = getPostConditionTitle(pc);
+  const iconString = imageCanonicalUri ?? getIconStringFromPostCondition(pc);
+  const pcTicker = getSymbolFromPostCondition(pc);
+  const pcAmount = getAmountFromPostCondition(pc);
+  const name = getNameFromPostCondition(pc);
+
+  const contractName = 'contractName' in pc.principal && pc.principal.contractName.content;
+  const address = 'address' in pc.principal ? addressToString(pc.principal.address) : '';
+  const isSending = address === currentAccount?.address;
+  const contractId = formatContractId(address, pc.asset.contractName.content);
+  const amount = asset?.decimals ? ftDecimals(pcAmount, asset.decimals) : pcAmount;
+  const ticker = asset?.symbol || pcTicker;
+
+  const isContractPrincipal =
+    !!contractName ||
+    (payload.payloadType === PayloadType.ContractCall &&
+      addressToString(payload.contractAddress) === address) ||
+    address.includes('.');
+
+  const message = pc.conditionCode
+    ? `${getPostConditionCodeMessage(
+        pc.conditionCode,
+        isSending
+      )} ${amount} ${ticker} or the transaction will abort.`
+    : undefined;
```

Comment:
would be nice if both of these were fns

Reply:
Sorry, I wasn't really spending time refactoring this code. I know there is duplicate code and not great. I was just relocating/isolating it from the legacy request and v1 was just hoping not to break anything displaying it in the Approver UX. I was then going to do a v2 looking at new designs and refactor this code. I can stop and do now if not critical to get this merged.

Reply:
Made a shared utils to format for the time being.